project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginStateObservable.kt

```diff
@@ -0,0 +1,60 @@
+package org.plan.research.minimization.plugin.settings
+
+import com.intellij.openapi.components.BaseState
+
+class MinimizationPluginStateObservable : BaseState() {
+    private var localState = MinimizationPluginState(AppSettings.getInstance())
+    var compilationStrategy = StateDelegate(
+        getter = { localState.compilationStrategy },
+        setter = { localState.compilationStrategy = it },
+    )
+    var temporaryProjectLocation = StateDelegate(
+        getter = { localState.temporaryProjectLocation },
+        setter = { localState.temporaryProjectLocation = it },
+    )
+    var snapshotStrategy = StateDelegate(
+        getter = { localState.snapshotStrategy },
+        setter = { localState.snapshotStrategy = it },
+    )
+    var exceptionComparingStrategy = StateDelegate(
+        getter = { localState.exceptionComparingStrategy },
+        setter = { localState.exceptionComparingStrategy = it },
+    )
+    var stages = StateDelegate(
+        getter = { localState.stages },
+        setter = { localState.stages = it },
+    )
+    var minimizationTransformations = StateDelegate(
+        getter = { localState.minimizationTransformations },
+        setter = { localState.minimizationTransformations = it },
+    )
+
+    fun updateSettingsState() {
```

Comment:
Let's remake this function. Instead of what we are doing now, we can block changes by the settings UI during action execution. It will be better I guess